front/pages/api/v1/w/[wId]/members/validate.ts

```diff
@@ -0,0 +1,85 @@
+import type { WithAPIErrorResponse } from "@dust-tt/types";
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import * as reporter from "io-ts-reporters";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { withPublicAPIAuthentication } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { MembershipResource } from "@app/lib/resources/membership_resource";
+import { UserResource } from "@app/lib/resources/user_resource";
+import { renderLightWorkspaceType } from "@app/lib/workspace";
+import { apiError } from "@app/logger/withlogging";
+
+export type ValidateMemberResponseBody = {
+  valid: boolean;
+};
+
+/**
+ * @ignoreswagger
+ * Validates an email corresponds to an active member in a specific workspace. For Dust managed apps only - undocumented.
+ */
+
+export const validateEmailSchema = t.type({
+  email: t.string,
+});
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<WithAPIErrorResponse<ValidateMemberResponseBody>>,
+  auth: Authenticator
+): Promise<void> {
+  const bodyValidation = validateEmailSchema.decode(req.body);
+
+  if (isLeft(bodyValidation)) {
+    const pathError = reporter.formatValidationErrors(bodyValidation.left);
+
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: `Invalid request body: ${pathError}`,
+      },
+    });
+  }
+
+  const { email } = bodyValidation.right;
+
+  switch (req.method) {
+    case "POST":
+      const user = await UserResource.fetchByEmail(email);
+
+      const workspace = renderLightWorkspaceType({
```

Comment:
you likely don't need that WorkspaceType is accepted by LightWorkspaceType

Reply:
👍